lib/app/migrate.ts

```diff
@@ -0,0 +1,584 @@
+import { CLIAccount } from '@hubspot/local-dev-lib/types/Accounts';
+import {
+  confirmPrompt,
+  inputPrompt,
+  listPrompt,
+  promptUser,
+} from '../prompts/promptUtils';
+import { ApiErrorContext, logError } from '../errorHandlers';
+import { EXIT_CODES } from '../enums/exitCodes';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  uiAccountDescription,
+  uiCommandReference,
+  uiLine,
+  uiLink,
+} from '../ui';
+import { i18n } from '../lang';
+import { isAppDeveloperAccount } from '../accountTypes';
+import { selectPublicAppPrompt } from '../prompts/selectPublicAppPrompt';
+import { fetchPublicAppMetadata } from '@hubspot/local-dev-lib/api/appsDev';
+import { createProjectPrompt } from '../prompts/createProjectPrompt';
+import { ensureProjectExists } from '../projects';
+import { trackCommandMetadataUsage } from '../usageTracking';
+import SpinniesManager from '../ui/SpinniesManager';
+import { handleKeypress } from '../process';
+import {
+  checkMigrationStatus,
+  downloadProject,
+  migrateApp as migrateNonProjectApp_v2023_2,
+  beginMigration,
+  finishMigration,
+  listAppsForMigration,
+} from '@hubspot/local-dev-lib/api/projects';
+import { poll } from '../polling';
+import path from 'path';
+import { getCwd, sanitizeFileName } from '@hubspot/local-dev-lib/path';
+import { getHubSpotWebsiteOrigin } from '@hubspot/local-dev-lib/urls';
+import { extractZipArchive } from '@hubspot/local-dev-lib/archive';
+import { ArgumentsCamelCase } from 'yargs';
+import { MigrateAppOptions } from '../../types/Yargs';
+import chalk from 'chalk';
+import { validateUid } from '@hubspot/project-parsing-lib';
+import { MigrationApp } from '@hubspot/local-dev-lib/types/Project';
+import { UNMIGRATABLE_REASONS } from '@hubspot/local-dev-lib/constants/projects';
+import { mapToUserFacingType } from '@hubspot/project-parsing-lib/src/lib/transform';
+
+function getUnmigratableReason(reasonCode: string) {
```

Comment:
Don't forget to add return types on these functions.

Reply:
Done